ndcube/mixins/ndslicing.py

```diff
@@ -40,22 +42,30 @@ def _slice(self, item):
         to add support for ``missing_axes`` and ``extra_coords`` and overwrites
         the astropy handling of wcs slicing.
         """
+
+        item = tuple(sanitize_slices(item, len(self.dimensions)))
         kwargs = super()._slice(item)
 
-        wcs, missing_axes = self._slice_wcs_missing_axes(item)
-        kwargs['wcs'] = wcs
-        kwargs['missing_axes'] = missing_axes
-        kwargs['extra_coords'] = self._slice_extra_coords(item, missing_axes)
+        # Store the original dimension of the wcs object before slicing
+        prev_dim = self.high_level_wcs.low_level_wcs.world_n_dim
 
+        # Sanitize the input arguments
+        wcs = self._slice_wcs(item)
+        print('ITEM {0}'.format(item))
+
+        # Set the kwargs values
+        kwargs['wcs'] = wcs
+        kwargs['extra_coords'] = self._slice_extra_coords(item, wcs._pixel_keep, prev_dim)
         return kwargs
 
-    def _slice_wcs_missing_axes(self, item):
-        # here missing axis is reversed as the item comes already in the reverse order
-        # of the input
-        return utils.wcs._wcs_slicer(
-            self.wcs, copy.deepcopy(self.missing_axes[::-1]), item)
+    def _slice_wcs(self, item):
+
```

Comment:
Add a docstring here.  Say what order the item should be.  Rename ```item``` to ```item_wcs``` or something similar to show what order the variable represents.

Reply:
The order should be in WCS order here.

Reply:
Sure, I will make a comment/change the variable name for the ordering.